src/psyclone/tests/domain/gocean/kernel/gokern_test.py

```diff
@@ -120,3 +124,97 @@ def test_gok_get_kernel_schedule():
             "PSyIR" in err_text)
     assert ("does not contain the routine that it names as implementing the "
             "kernel ('compute_cu_code')" in err_text)
+
+
+# -----------------------------------------------------------------------------
+def test_gok_reference_accesses(fortran_writer):
+    '''Test that GOcean kernel information are correct, and that the index
+    information are PSyIR nodes.
+
+    '''
+    # Large stencil has 123,110,100
+    _, invoke = get_invoke("large_stencil.f90", "gocean1.0", idx=0)
+    schedule = invoke.schedule
+
+    # Get the first kernel
+    kern1 = schedule.walk(GOKern)[0]
+    vai = VariablesAccessInfo(kern1)
+    assert str(vai) == "cu_fld: WRITE, p_fld: READ, u_fld: READ"
+    p_fld = vai[Signature("p_fld")]
+    # We can't have lists in a set, so we convert the lists to string
+    # for easy comparison. Calling `to_fortran` also ensures that the
+    # component indices are PSyIR nodes (not strings)
+
+    # Convert each PSyIR index into a string, and then also convert each
+    # index pair into a string to be added to the set (we can't have lists in
+    # in a set, so that's the easiest way to compare them)
+    result = set()
+    for access in p_fld:
+        for indices in access.component_indices:
+            result.add(str([fortran_writer(index) for index in indices]))
+
+    # The stencil is 123, 110, 100 - test that appropriate accesses were
```

Comment:
I avoided top/middle/bottom in the first place, since I think that might actually depend on how you use it (??), i.e. just a convention.

But I've changed the wording of the comment, adding "is declared as", i.e. this is the order which is used in the kernel meta data.

Reply:
Well, it is a convention but it is the GOcean convention and this is a gocean test so I think we should be consistent. Perhaps add a link to the relevant bit of the docs too?

Reply:
OK, I am not sure if I get it, but I've changed the comment to `The stencil is 100, 110, 123 - test ... ` (and did the same on line 135). Is this what you wanted?

Reply:
Or do you also want me the order of the following line which verify the stencil? I.e. atm we have:
```
    expected = {
        # First stencil direction of 123: 1
        "['i - 1', 'j - 1']",
        # Second stencil direction of 123: 2
   ...
```
?

Reply:
I don't mind what order the code does its checks so that's fine. Thanks for updating the comments.